src/scaffoldmaker/meshtypes/meshtype_3d_lung2.py

```diff
@@ -0,0 +1,2139 @@
+        baseLungTerms = ["left lung", "right lung", "right lung accessory lobe"]
+        lobe = {}
+
+        for term in surfaceTerms:
+            if (numberOfLeftLung == 1) and (term in subLeftLungTerms):
+                continue
+            if (hasAccessoryLobe is False) and (term == "right lung accessory lobe"):
+                continue
+
+            group = getAnnotationGroupForTerm(annotationGroups, get_lung_term(term))
+            group2d = group.getFieldElementGroup(mesh2d)
+
+            if "lobe of" in term:
+                lobe.update({term: group2d})
+
+            group2d_exterior = fm.createFieldAnd(group2d, is_exterior)
+            surfaceGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, get_lung_term(term + " surface"))
+            surfaceGroup.getMeshGroup(mesh2d).addElementsConditional(group2d_exterior)
+
+            if term in baseLungTerms:
+                if term == "right lung accessory lobe":
+                    group2D_exterior_bottom = fm.createFieldAnd(group2d_exterior, is_xi3_0)
+                else:
+                    group2d_exterior_xi1_0 = fm.createFieldAnd(group2d_exterior, is_xi1_0)
+                    group2d_exterior_xi1_1 = fm.createFieldAnd(group2d_exterior, is_xi1_1)
+                    group2d_exterior_xi1_01 = fm.createFieldOr(group2d_exterior_xi1_0, group2d_exterior_xi1_1)
+                    group2D_exterior_bottom = fm.createFieldXor(group2d_exterior, group2d_exterior_xi1_01)
+
+                baseSurfaceGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, get_lung_term("base of " + term + " surface"))
+                baseSurfaceGroup.getMeshGroup(mesh2d).addElementsConditional(group2D_exterior_bottom)
+
+        # Fissures
+        fissureTerms = ["oblique fissure of right lung", "horizontal fissure of right lung", "oblique fissure of left lung"]
+        for fissureTerm in fissureTerms:
+            if (fissureTerm == "oblique fissure of left lung") and (numberOfLeftLung == 2):
+                fissureGroup = fm.createFieldAnd(lobe["upper lobe of left lung"], lobe["lower lobe of left lung"])
+            elif fissureTerm == "oblique fissure of right lung":
+                fissureGroup = fm.createFieldAnd(fm.createFieldOr(lobe["middle lobe of right lung"], lobe["upper lobe of right lung"]),
+                                                        lobe["lower lobe of right lung"])
+            elif fissureTerm == "horizontal fissure of right lung":
+                fissureGroup = fm.createFieldAnd(lobe["upper lobe of right lung"], lobe["middle lobe of right lung"])
+            fissureSurfaceGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, get_lung_term(fissureTerm))
+            fissureSurfaceGroup.getMeshGroup(mesh2d).addElementsConditional(fissureGroup)
+
+def createLungNodes(spaceFromCentre, lengthUnit, widthUnit, heightUnit, fissureAngle, obliqueProportion,
```

Comment:
No. The following functions are not shared with lung1. They are completely different since lung2 generates nodes from the defined parameters while Lung1 is much simpler creating nodes without any parameters.

Reply:
That is true for node parameters.
However, I often separate code into 3 stages: making parameter arrays, building nodes, building elements, and the latter 2 steps should be the same (for the case without open fissures).
I'm not going to press for this now unless you can see obvious common functions.
Generally, unless code is trivial or the differences are throughout a piece of code (turning it into "spaghetti", we do our best to put it into common functions when possible.